sockeye/inference.py

```diff
@@ -1278,13 +1279,16 @@ def _beam_search(self,
 
             # (3) get beam_size winning hypotheses for each sentence block separately
             # TODO(fhieber): once mx.nd.topk is sped-up no numpy conversion necessary anymore.
-            scores = scores.asnumpy()  # convert to numpy once to minimize cross-device copying
+            if self.context == mx.cpu():
+                scores = scores.asnumpy()  # convert to numpy once to minimize cross-device copying
```

Comment:
minor suggestion: we could avoid this if branch by pulling it into utils.smallest_k, so that both methods expect an mxnet.NDArray.

Reply:
not sure this would be beneficial as we would then numpify slices of the large tensor.